### PR TITLE
Fix webhook conflict that occurs when reinstalling after the default installation has failed

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
This is a supplement to https://github.com/istio/istio/pull/48700, covering the last missing case regarding webhook conflict. This fix is to ignore the in-cluster `istio-sidecar-injector` webhook in webhook analysis if the installation is a default one, as it will be deactivated upon successful installation. Otherwise if that webhook is activated due to failed installation before, which applied the webhook manifest and didn't do `ProcessDefaultWebhook`, the installation can still be conflict.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/config/analysis/analyzers/webhook/webhook.go [hunk 1]
```diff
@@ -73,6 +74,11 @@ func (a *Analyzer) Analyze(context analysis.Context) {
 	resources := map[string]*resource.Instance{}
 	revisions := sets.New[string]()
 	context.ForEach(gvk.MutatingWebhookConfiguration, func(resource *resource.Instance) bool {
+		if a.SkipDefaultRevisionedWebhook {
+			if isDefaultRevisionedWebhook(resource.Message.(*v1.MutatingWebhookConfiguration)) {
+				return true
+			}
+		}
```
Comment (zirain) [inline pkg/config/analysis/analyzers/webhook/webhook.go:81]: a.SkipDefaultRevisionedWebhook && isDefaultRevisionedWebhook(resource.Message.(*v1.MutatingWebhookConfiguration)) ？

Comment (hanxiaop) [reply pkg/config/analysis/analyzers/webhook/webhook.go]: @zirain Fixed, thanks

**Review by zirain — APPROVED** [review]

**Comment by hanxiaop** [comment]
@istio/wg-networking-maintainers PTAL, thanks

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**